examples/coupled_quartic_hamiltonian.py

```diff
@@ -0,0 +1,656 @@
+    #
+    #
+    #eqPt = 1
+    #eqPt = get_eq_pts_coupled(eqNum, par)
+    #evalue, evector = np.linalg.eig(jacobian_coupled([eqPt[0],eqPt[1],0,0],par))
+    #evector = RemoveInfinitesimals(evector[:,2])
+    #correcx = (evector[0]*1j).real
+    #correcy = (evector[1]*1j).real
+    correcx = (par[1]*par[6])/(-evaluelamb**2 - par[3] + par[6])
+    correcy = par[1]
+    
+    return correcx, correcy
+
+
+def guess_lin_coupled(eqPt, Ax, par):
+    """ Returns an initial guess for the unstable periodic orbit """ 
+    
+    correcx, correcy = eigvector_coupled(par)
+    
+
+    return [eqPt[0]-Ax*correcx,eqPt[1]-Ax*correcy,0,0]
+
+
+def jacobian_coupled(eqPt, par):
+    """ Returns Jacobian of the Hamiltonian vector field """
+    
+    x,y,px,py = eqPt[0:4]
+    
+    # The first order derivative of the Hamiltonian.
+    dVdx = (-par[3]+par[6])*x+par[4]*x**3-par[6]*y
+    dVdy = (par[5]+par[6])*y-par[6]*x
+
+    # The following is the Jacobian matrix 
+    d2Vdx2 = -par[3]+par[6]+par[4]*3*x**2
+        
+    d2Vdy2 = par[5]+par[6]
+
+    d2Vdydx = -par[6]
+        
+    d2Vdxdy = d2Vdydx    
+
+    Df = np.array([[  0,     0,    par[0],    0],
+                   [0,     0,    0,    par[1]],
+                   [-d2Vdx2,  -d2Vdydx,   0,    0],
+                   [-d2Vdxdy, -d2Vdy2,    0,    0]])
+    
+    return Df
+
+
+def varEqns_coupled(t,PHI,par):
+    """    
+    Returns the state transition matrix , PHI(t,t0), where Df(t) is the Jacobian of the 
+    Hamiltonian vector field
+    
+    d PHI(t, t0)
+    ------------ =  Df(t) * PHI(t, t0)
+        dt
+    
+    """
+    
+    phi = PHI[0:16]
+    phimatrix  = np.reshape(PHI[0:16],(4,4))
+    x,y,px,py = PHI[16:20]
+    
+    
+    # The first order derivative of the potential energy.
+    dVdx = (-par[3]+par[6])*x+par[4]*x**3-par[6]*y
+    dVdy = (par[5]+par[6])*y-par[6]*x
+
+    # The second order derivative of the potential energy. 
+    d2Vdx2 = -par[3]+par[6]+par[4]*3*x**2
+        
+    d2Vdy2 = par[5]+par[6]
+
+    d2Vdydx = -par[6]
+
+    
+    d2Vdxdy = d2Vdydx    
+
+    Df    = np.array([[  0,     0,    par[0],    0],
+              [0,     0,    0,    par[1]],
+              [-d2Vdx2,  -d2Vdydx,   0,    0],
+              [-d2Vdxdy, -d2Vdy2,    0,    0]])
+
+    
+    phidot = np.matmul(Df, phimatrix) # variational equation
+
+    PHIdot        = np.zeros(20)
+    PHIdot[0:16]  = np.reshape(phidot,(1,16)) 
+    PHIdot[16]    = px/par[0]
+    PHIdot[17]    = py/par[1]
+    PHIdot[18]    = -dVdx 
+    PHIdot[19]    = -dVdy
+    
+    return list(PHIdot)
+
+
+def diffcorr_setup_coupled():
+    """ 
+    Returns settings for differential correction method 
+        
+    Settings include choosing coordinates for event criteria, convergence criteria, and 
+    correction (see references for details on how to choose these coordinates).
+    """
+    
+    dxdot1 = 1
+    correctx0 = 0
+    MAXdxdot1 = 1.e-10
+    drdot1 = dxdot1
+    correctr0 = correctx0
+    MAXdrdot1 = MAXdxdot1

```

Comment:
These seem like they should be provided as keyword arguments to some other function, especially since they're all constants. If the names of the settings would vary between specific problems, can you find a more general name for them?